### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by orgads**
Resolved most of the warnings. Some still exist.

### CODE REVIEW

File: src/test/java/hudson/matrix/MatrixProjectTest.java [hunk 1]
```diff
@@ -569,7 +569,18 @@ public void dontRunOnExclusiveSlave() throws Exception {
             projects.add(m);
         }
 
-        DumbSlave s = new DumbSlave("big", "this is a big slave", j.createTmpDir().getPath(), "20", EXCLUSIVE, "", j.createComputerLauncher(null), RetentionStrategy.NOOP);
+        TemporaryFolder tf = new TemporaryFolder();
```
Comment (jvz) [inline src/test/java/hudson/matrix/MatrixProjectTest.java]: This won't work properly unless you make it a `@Rule` field.

Comment (olivergondza) [reply src/test/java/hudson/matrix/MatrixProjectTest.java]: Good catch!

**Comment by orgads** [comment]
> Please fix the problem reported by @jvz

Done

**Comment by olivergondza** [comment]
Thanks!

File: src/main/java/hudson/tasks/test/MatrixTestResult.java [hunk 2]
```diff
@@ -52,12 +52,12 @@ public MatrixTestResult() {}
      */
     @Override
     protected String getChildName(AbstractTestResultAction tr) {
-        return tr.owner.getProject().getName();
+        return ((MatrixBuild)tr.run).getProject().getName();
```
Comment (jglick) [inline src/main/java/hudson/tasks/test/MatrixTestResult.java:55]: https://ci.jenkins.io/job/Tools/job/bom/job/PR-251/1/testReport/junit/hudson.tasks.junit/JUnitResultArchiverTest/pct_junit_2_222_x___specialCharsInRelativePath/

**Pull request merged**